packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/reducer/index.js

```diff
@@ -43,12 +43,12 @@ export function reducer(state = initialState, action) {
         const { stageId } = payload;
         const { currentWorkflow } = state.clientState;
 
-        draft.clientState.currentWorkflow.data = {
-          ...currentWorkflow.data,
-          stages: currentWorkflow.data.stages.filter(
-            (stage) => (stage?.id ?? stage.__temp_key__) !== stageId
-          ),
-        };
+        draft.clientState.currentWorkflow.data.stages = currentWorkflow.data.stages.filter(
+          (stage) => (stage?.id ?? stage.__temp_key__) !== stageId
+        );
+
+        draft.clientState.currentWorkflow.hasDeletedServerStages =
+          !!state.serverState.currentWorkflow.stages.find((stage) => stage.id === stageId);
 
         break;
```

Comment:
What about the case where the user:

 - Adds a new stage
 - Deletes a different already existing stage
 - Deletes the new stage you just added
 - Clicks save

 In this case I think `hasDeletedServerStages` will be set to false and no warning will be displayed, when really you have deleted one of the server stages

Reply:
Done in: https://github.com/strapi/strapi/pull/15747/commits/00bffbd079e4396d2b46e200138985b4484e4815

Do you think that would work?